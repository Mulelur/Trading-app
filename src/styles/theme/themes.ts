const lightTheme = {
  primary: '#12161E',
  text: '#12161E',
  textSecondary: 'rgb(102, 191, 191)',
  background: 'rgb(234, 246, 246)',
  backgroundVariant: 'rgb(102, 191, 191)',
  border: 'rgb(102, 191, 191)',
  borderLight: 'rgb(196, 212, 237, .3)',
};

const darkTheme: Theme = {
  primary: 'rgba(220,120,95,1)',
  text: 'rgba(241,233,231,1)',
  textSecondary: 'rgba(241,233,231,0.6)',
  background: 'rgba(0,0,0,1)',
  backgroundVariant: 'rgba(28,26,26,1)',
  border: 'rgba(241,233,231,0.15)',
  borderLight: '#2c2525',
};

export type Theme = typeof lightTheme;

export const themes = {
  light: lightTheme,
  dark: darkTheme,
};
