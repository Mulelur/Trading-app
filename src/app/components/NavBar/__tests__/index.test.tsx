import * as React from 'react';
import { ThemeProvider } from 'styles/theme/ThemeProvider';

import { NavBar } from '../index';
import { render } from '@testing-library/react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Store } from '@reduxjs/toolkit';
import { configureAppStore } from 'store/configureStore';

const renderNavBar = (store: Store) =>
  render(
    <Provider store={store}>
      <BrowserRouter>
        <ThemeProvider>
          <Routes>
            <Route path="*" element={<NavBar />} />
          </Routes>
        </ThemeProvider>
      </BrowserRouter>
    </Provider>,
  );

describe('<NavBar />', () => {
  let renderedOutput: ReturnType<typeof renderNavBar>;

  beforeEach(() => {
    const store = configureAppStore();
    renderedOutput = renderNavBar(store);
  });

  it('should match snapshot', () => {
    expect(renderedOutput).toMatchSnapshot();
  });
});
