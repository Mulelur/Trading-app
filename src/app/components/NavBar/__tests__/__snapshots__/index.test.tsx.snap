// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
  color: rgba(215,113,88,1);
}

.c7 {
  color: rgba(215,113,88,1);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.15rem 0.3rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:hover {
  opacity: 0.8;
}

.c7:active {
  opacity: 0.4;
}

.c7 .icon {
  margin-right: 0.2rem;
}

.c1 {
  display: none;
}

.c2 {
  color: rgba(215,113,88,1);
  padding-right: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: 1.25rem;
  color: rgba(58,52,51,1);
  font-weight: bold;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  padding-right: 0.5rem;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
  width: 100%;
  padding-left: 10rem;
  padding-right: 10rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media only screen and (max-width:768px) {
  .c1 {
    display: block;
  }
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    padding-left: 1rem;
    padding-right: 3rem;
  }
}

@media only screen and (min-width:600px) {
  .c0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding-left: 17rem;
    padding-right: 7rem;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    padding-left: 18rem;
    padding-right: 8rem;
  }
}

<body>
    <div>
      <div
        class="c0"
      >
        <div
          class="c1"
        >
          <div
            class=""
          >
            <div
              class="c2"
            >
              <svg>
                menu-icon.svg
              </svg>
            </div>
          </div>
        </div>
        <div
          class="c3"
        >
          <a
            class="c4"
            href="/"
          >
            <div
              class="c5"
            >
              <svg>
                logo-icon.svg
              </svg>
            </div>
            Weblinnk
          </a>
        </div>
        <div
          class="c6"
        >
          <a
            class="c7"
            href="/profile"
            rel="noopener noreferrer"
            target="_blank"
            title="Documentation Page"
          >
            <svg>
              account-icon.svg
            </svg>
          </a>
          <a
            class="c7"
            href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
            rel="noopener noreferrer"
            target="_blank"
            title="Github Page"
          >
            <svg>
              bell-icon.svg
            </svg>
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": .c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: -1rem;
  color: rgba(215,113,88,1);
}

.c7 {
  color: rgba(215,113,88,1);
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0.15rem 0.3rem;
  font-size: 0.875rem;
  font-weight: 500;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c7:hover {
  opacity: 0.8;
}

.c7:active {
  opacity: 0.4;
}

.c7 .icon {
  margin-right: 0.2rem;
}

.c1 {
  display: none;
}

.c2 {
  color: rgba(215,113,88,1);
  padding-right: 0.8rem;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 4rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  font-size: 1.25rem;
  color: rgba(58,52,51,1);
  font-weight: bold;
  margin-right: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 {
  padding-right: 0.5rem;
}

.c0 {
  box-shadow: 0 1px 0 0 rgba(58,52,51,0.05);
  height: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  background-color: rgba(255,255,255,1);
  z-index: 2;
  width: 100%;
  padding-left: 10rem;
  padding-right: 10rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

@media only screen and (max-width:768px) {
  .c1 {
    display: block;
  }
}

@supports (-webkit-backdrop-filter:blur(10px)) or (backdrop-filter:blur(10px)) {
  .c0 {
    -webkit-backdrop-filter: blur(10px);
    backdrop-filter: blur(10px);
    background-color: rgba(255,255,255,0.75);
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    padding-left: 1rem;
    padding-right: 3rem;
  }
}

@media only screen and (min-width:600px) {
  .c0 {
    padding-left: 5rem;
    padding-right: 5rem;
  }
}

@media only screen and (min-width:768px) {
  .c0 {
    padding-left: 17rem;
    padding-right: 7rem;
  }
}

@media only screen and (min-width:992px) {
  .c0 {
    padding-left: 18rem;
    padding-right: 8rem;
  }
}

<div>
    <div
      class="c0"
    >
      <div
        class="c1"
      >
        <div
          class=""
        >
          <div
            class="c2"
          >
            <svg>
              menu-icon.svg
            </svg>
          </div>
        </div>
      </div>
      <div
        class="c3"
      >
        <a
          class="c4"
          href="/"
        >
          <div
            class="c5"
          >
            <svg>
              logo-icon.svg
            </svg>
          </div>
          Weblinnk
        </a>
      </div>
      <div
        class="c6"
      >
        <a
          class="c7"
          href="/profile"
          rel="noopener noreferrer"
          target="_blank"
          title="Documentation Page"
        >
          <svg>
            account-icon.svg
          </svg>
        </a>
        <a
          class="c7"
          href="https://github.com/react-boilerplate/react-boilerplate-cra-template"
          rel="noopener noreferrer"
          target="_blank"
          title="Github Page"
        >
          <svg>
            bell-icon.svg
          </svg>
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
